### PR TITLE
Add 0.66.0 CHANGELOG entry

### PR DESCRIPTION
**Pull request by ArthurSens**
## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [X] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,25 @@
+## 0.66.0 / 2023-06-14
+
+After research, we noticed how the Config-Reloader container is over dimensioned regarding it's CPU resources. We decided to change the default CPU requirements from 100m to 10m. For most environments this won't make a difference, but if you still need to increase Config-Reloader CPU resources, you can do so by adding the flags `-config-reloader-cpu-request` and `-config-reloader-cpu-limit` to the operator container.
```
Comment (simonpasquier) [inline CHANGELOG.md:3]: (suggestion)
```suggestion
After research, we noticed how the default CPU requests of the config-reloader container were too high compared to the effective usage. We decided to decrease the default CPU requests from `100m` to `10m`. For most environments, this won't make a difference but if you need to increase the values, you can use the `-config-reloader-cpu-request` and `-config-reloader-cpu-limit` CLI arguments of the Prometheus operator. 
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,25 @@
+## 0.66.0 / 2023-06-14
+
+After research, we noticed how the Config-Reloader container is over dimensioned regarding it's CPU resources. We decided to change the default CPU requirements from 100m to 10m. For most environments this won't make a difference, but if you still need to increase Config-Reloader CPU resources, you can do so by adding the flags `-config-reloader-cpu-request` and `-config-reloader-cpu-limit` to the operator container. 
+
+* [CHANGE] Config-Reloader default CPU requirements lowered from 100m to 10m #5539
```
Comment (simonpasquier) [inline CHANGELOG.md:5]:
```suggestion
* [CHANGE] Decrease the default CPU requests for the config-reloader container from `100m` to `10m`. #5539
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,25 @@
...
+* [FEATURE] Add `spec.alertmanagerConfiguration.global.PagerdutyURL` to the `Alertmanager` CRD #5469
```
Comment (simonpasquier) [inline CHANGELOG.md:6]: (nit) missing final dot. Same remark for the other entries.

```suggestion
* [FEATURE] Add `spec.alertmanagerConfiguration.global.PagerdutyURL` to the `Alertmanager` CRD. #5469
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,25 @@
...
+* [FEATURE] Add `spec.valumeMounts` to the `ThanosRuler` CRD #5541
+* [FEATURE] Add `spec.remoteWrite.sendNativeHistograms` to `Prometheus` and `PrometheusAgent` CRDs #5564
+* [FEATURE] Add `spec.tracingConfig` to `Prometheus` and `PrometheusAgent` CRDs #5591 #5640
+* [FEATURE] Add `followRedirects` in order to allow to change the default value in `RemoteRead` #5612
```
Comment (simonpasquier) [inline CHANGELOG.md:10]:
```suggestion
* [FEATURE] Add `followRedirects` field to the `RemoteRead` configuration. #5612
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,25 @@
...
+* [FEATURE] Add option to configure `automountServiceAccountToken` for `Alertmanager` CRD #5474
```
Comment (simonpasquier) [inline CHANGELOG.md:11]:
```suggestion
* [FEATURE] Add `spec.automountServiceAccountToken` to the `Alertmanager` CRD #5474
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,25 @@
...
+* [FEATURE] Add `keep_firing_for` to rules in `PrometheusRule` CRD #5651
```
Comment (simonpasquier) [inline CHANGELOG.md:12]:
```suggestion
* [FEATURE] Support `keep_firing_for` fields for `PrometheusRule` CRD. #5651
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,25 @@
...
+* [FEATURE] Allow to add custom annotations to resources managed by operator #5626
```
Comment (simonpasquier) [inline CHANGELOG.md:13]: (nit)
```suggestion
* [FEATURE] Allow to add custom annotations to all resources managed by the operator. #5626
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,25 @@
...
+* [FEATURE] Add `BasicAuth` and `Authorization` support to `ScrapeConfig` CRD #5642
+* [ENHANCEMENT] Add label selector to filter nodes monitored by the Operator #5641
```
Comment (simonpasquier) [inline CHANGELOG.md:15]:
```suggestion
* [ENHANCEMENT] Add `-kubelet-selector` CLI argument to filter the Kubernetes nodes by labels. #5641
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,25 @@
...
+* [BUGFIX] Fix bug with logs that don't end with new line(`\n`) characters. #5566
+* [BUGFIX] Prometheus and Alertmanager now pickup certificates after renewal #5535
+* [BUGFIX] Config-Reloader now gracefully handles SIGTERM signals #5617
```
Comment (simonpasquier) [inline CHANGELOG.md:18]:
```suggestion
* [BUGFIX] Fix config-reloader not handling SIGTERM signal. #5617
```

**Review by simonpasquier — COMMENTED** [review]
Thanks for compiling the list! Added a few suggestions/remarks but looks good overall :)

**Comment by ArthurSens** [comment]
Thanks for the initial look, should be ready for another review!

**Review by philipgough — APPROVED** [review]
lgtm

**Review by simonpasquier — APPROVED** [review]
:partying_face:

**Pull request merged**